src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -955,9 +953,11 @@ def _create_identity_instance(cli_ctx, *args, **kwargs):
     # PREVIEW: In Azure Stack environment, use core.instance_discovery=false to disable MSAL's instance discovery.
     instance_discovery = cli_ctx.config.getboolean('core', 'instance_discovery', True)
 
-    return Identity(*args, encrypt=encrypt, use_msal_http_cache=use_msal_http_cache,
+    return Identity(authority, tenant_id=tenant_id, client_id=client_id,
```

Comment:
Actually, custom client ID has been supported by `azure.cli.core.auth` during the MSAL migration long ago (https://github.com/Azure/azure-cli/pull/19853). `Identity` takes the `client_id=None` keyword argument since the first day, but this feature is never exposed.